icu4c/source/test/intltest/ssearch.cpp

```diff
@@ -524,40 +527,48 @@ UBool OrderList::matchesAt(int32_t offset, const OrderList &other) const
     return true;
 }
 
-static char *printOffsets(char *buffer, size_t n, OrderList &list)
+static char* printOffsets(std::string &buffer, OrderList &list)
 {
     int32_t size = list.size();
-    char *s = buffer;
+    buffer.clear();
+    buffer.reserve(size * 10 + 2);
```

Comment:
Is this some kind of performance optimization, to avoid having to dynamically resize the buffer? If so, I suggest removing it, this is test code, the possible performance gain is irrelevant compared to having a few extra lines of code to maintain.